src/synth/fluid_synth.c

```diff
@@ -5298,12 +5298,24 @@ fluid_synth_alloc_voice_LOCAL(fluid_synth_t *synth, fluid_sample_t *sample, int
     */
     {
         int mono = fluid_channel_is_playing_mono(channel);
-        fluid_mod_t *default_mod = synth->default_mod;
+        int override_default_modulators = FALSE;
+        fluid_mod_t *default_mod;
+        if (sample->default_modulators != NULL)
+        {
+            default_mod = sample->default_modulators;
+            override_default_modulators = TRUE;
+        }
+        else
+        {
+            default_mod = synth->default_mod;
+        }
 
         while(default_mod != NULL)
         {
             if(
-                /* See if default_mod is the velocity_to_attenuation modulator */
+                /* if SF explicitly wants the default vel2att modulator, keep it*/
+                override_default_modulators != TRUE &&
```

Comment:
This is an implementation detail I'm not too sure about. When fluidsynth has been configured for using a breath controller for triggering notes, it replace the default velocity to attenuation modulator. IMO, this logic should remain unchanged, i.e. if the soundfont contains a vel-to-atten modulator in its DMOD chunk and fluidsynth is in breath mode, it should still be replaced. Otherwise the breath logic of triggering and attenuating notes would be broken.